src/kernel.jl

```diff
@@ -247,7 +247,7 @@ function set_arg!{T}(k::Kernel, idx::Integer, arg::T)
     ref = if isbits(T_aligned)
         Base.RefValue(packed)
     else
-        packed
+        pointer_from_objref(packed)
```

Comment:
Wouldn't it be fine to just say: `Ref(packed)`? and then let the logic for references take over instead of re-implementing it here?

Reply:
Yeah, that might just work...

Reply:
Do you want to quickly fix it otherwise I will merge this and tag a point release.